internal/cli/atlas/deployments/connect.go

```diff
@@ -17,120 +17,58 @@ package deployments
-			return err
-		}
-	}
-
-	return nil
-}
-
-func (opts *ConnectOpts) Run(ctx context.Context) error {
-	if err := opts.podmanClient.Ready(ctx); err != nil {
-		return err
-	}
-
-	telemetry.AppendOption(telemetry.WithDeploymentType(options.LocalCluster)) // always local
-	if opts.DeploymentName != "" {
-		if err := opts.DeploymentOpts.CheckIfDeploymentExists(ctx); err != nil {
-			return err
-		}
-	} else {
-		if err := opts.DeploymentOpts.Select(ctx); err != nil {
-			return err
-		}
-	}
-
-	if opts.connectWith == "" {
-		var err error
-		if opts.connectWith, err = opts.DeploymentOpts.PromptConnectWith(); err != nil {
-			return err
-		}
-	}
-
-	connectionString, err := opts.ConnectionString(ctx)
-	if err != nil {
-		return err
-	}
-
-	switch opts.connectWith {
-	case options.ConnectWithConnectionString:
-		opts.Print(connectionString)
-	case compassConnect:
-		if !compass.Detect() {
-			return errCompassNotInstalled
-		}
-		if _, err := log.Warningln("Launching MongoDB Compass..."); err != nil {
-			return err
-		}
-		return compass.Run("", "", connectionString)
-	case mongoshConnect:
-		if !mongosh.Detect() {
-			return errMongoshNotInstalled
-		}
-		return mongosh.Run("", "", connectionString)
-	}
-
-	return nil
+func Run(ctx context.Context, opts *options.ConnectOpts) error {
+	return opts.Connect(ctx)
 }
 
 // atlas deployments connect [clusterName].
 func ConnectBuilder() *cobra.Command {
-	opts := &ConnectOpts{}
+	opts := &options.ConnectOpts{}
 	cmd := &cobra.Command{
 		Use:     "connect [deploymentName]",
 		Short:   "Connect to a deployment.",
 		Args:    require.MaximumNArgs(1),
-		GroupID: "local",
+		GroupID: "all",
 		Annotations: map[string]string{
 			"deploymentNameDesc": "Name of the deployment that you want to connect to.",
 		},
 		PreRunE: func(cmd *cobra.Command, args []string) error {
 			if len(args) == 1 {
 				opts.DeploymentName = args[0]
 			}
-			opts.podmanClient = podman.NewClient(log.IsDebugLevel(), log.Writer())
-			return opts.PreRunE(opts.InitOutput(cmd.OutOrStdout(), ""), opts.InitStore(opts.podmanClient), opts.validateFlags)
+			return opts.PreRunE(
+				opts.InitOutput(cmd.OutOrStdout(), ""),
+				opts.InitStore(podman.NewClient(log.IsDebugLevel(), log.Writer())),
+				opts.InitAtlasStore(cmd.Context()),
+			)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return opts.Run(cmd.Context())
+			return Run(cmd.Context(), opts)
 		},
 	}
 
-	cmd.Flags().StringVar(&opts.connectWith, flag.ConnectWith, "", usage.ConnectWith)
+	cmd.Flags().StringVar(&opts.ConnectWith, flag.ConnectWith, "", usage.ConnectWith)
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
+	cmd.Flags().StringVar(&opts.DeploymentType, flag.TypeFlag, "", usage.DeploymentType)
+	cmd.Flags().StringVar(&opts.DBUsername, flag.Username, "", usage.DBUsername)
+	cmd.Flags().StringVar(&opts.DBUserPassword, flag.Password, "", usage.Password)
```

Comment:
I don't see it but does this support providing the password via stdin?

Reply:
yes, replied here https://github.com/mongodb/mongodb-atlas-cli/pull/2320/files#r1338813122